cli/azd/pkg/tools/github/github.go

```diff
@@ -197,6 +197,23 @@ func (cli *Cli) Login(ctx context.Context, hostname string) error {
 	return nil
 }
 
+// ApiCall uses gh cli to call https://api.<hostname>/<path>.
+func (cli *Cli) ApiCall(ctx context.Context, hostname, path string, headers []string) (string, error) {
```

Comment:
Intead of exposing `headers` as a top level param consider creating an `Options` object where additional optional parameters can be set.